Corona-Warn-App/src/deviceForTesters/java/de/rki/coronawarnapp/test/crash.ui/SettingsCrashReportViewModel.kt

```diff
@@ -0,0 +1,64 @@
+package de.rki.coronawarnapp.test.crash.ui
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.asLiveData
+import androidx.lifecycle.map
+import androidx.lifecycle.viewModelScope
+import com.squareup.inject.assisted.AssistedInject
+import de.rki.coronawarnapp.bugreporting.event.BugEvent
+import de.rki.coronawarnapp.bugreporting.reportProblem
+import de.rki.coronawarnapp.bugreporting.storage.repository.BugRepository
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModel
+import de.rki.coronawarnapp.util.viewmodel.SimpleCWAViewModelFactory
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import timber.log.Timber
+import java.lang.Exception
+
+class SettingsCrashReportViewModel @AssistedInject constructor(
+    private val crashReportRepository: BugRepository
+) : CWAViewModel() {
+
+    val crashReports = crashReportRepository.getAll().asLiveData()
+
+    private val selectedCrashReportMutable: MutableLiveData<BugEvent> by lazy { MutableLiveData() }
```

Comment:
I prefer to have public vals first, private ones afterward

Reply:
Package by feature, not by layer. As `crashReports` and ` selectedCrashReportMutable` are independent, the order doesn't really matter as within the  "package by feature" approach, it will never be possible to have always all `private` ones first.

```kotlin
private val internaCoffee : Flow
val coffee: Flow

private val internalBanana: Flow
val bananana: Flow
```